### PR TITLE
fix: cluster configuration validation for bool type

### PR DESCRIPTION
**Pull request by Ygnas**
# Issue link

[https://issues.redhat.com/browse/RHOAIENG-11022](https://issues.redhat.com/browse/RHOAIENG-11022)

# What changes have been made

Fixed cluster configuration validation for bool types, and changed so all type mismatches are reported simultaneously, rather than stopping at the first occurrence.

# Verification steps

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk/ray/cluster/config.py [hunk 1]
```diff
@@ -242,13 +242,15 @@ def _memory_to_resource(self):
 
     def _validate_types(self):
         """Validate the types of all fields in the ClusterConfiguration dataclass."""
+        errors = []
```
Comment (ChristianZaccaria) [inline src/codeflare_sdk/ray/cluster/config.py:245]: You could remove this initialisation, and instead below use:
```
errors = [f"'{field_info.name}' should be of type {expected_type}."]
```

**Review by ChristianZaccaria — COMMENTED** [review]
Small nitpick, but looks good! I'll give it a test tomorrow.

File: src/codeflare_sdk/ray/cluster/config.py [hunk 2]
```diff
@@ -242,13 +242,15 @@ def _memory_to_resource(self):
...
             value = getattr(self, field_info.name)
             expected_type = field_info.type
             if not self._is_type(value, expected_type):
-                raise TypeError(
-                    f"'{field_info.name}' should be of type {expected_type}"
-                )
+                errors.append(f"'{field_info.name}' should be of type {expected_type}.")
+
+        if errors:
+            raise TypeError("Type validation failed:\n" + "\n".join(errors))
```
[the remark this answers is not in the document]
Comment (Ygnas) [reply src/codeflare_sdk/ray/cluster/config.py:253]: That would pass all errors as a list without any spacing or new lines after each error?

File: src/codeflare_sdk/ray/cluster/config.py (hunk 1 above)
Comment (ChristianZaccaria) [reply src/codeflare_sdk/ray/cluster/config.py:245]: Correct, good spot, thanks Kevin

**Review by Bobbins228 — COMMENTED** [review]
Tested it out and works great 
/lgtm

File: src/codeflare_sdk/ray/cluster/config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KPostOffice) [reply src/codeflare_sdk/ray/cluster/config.py:253]: Yeah. I think I lean towards just line counting. I was thinking about it from a testing perspective, but from actual user experience, a pretty output with newlines and such is definitely nicer.

**Review by KPostOffice — APPROVED** [review]
/lgtm

**Pull request merged**